transport/src/main/java/io/zeebe/transport/impl/RequestContext.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.0. You may not use this file
+ * except in compliance with the Zeebe Community License 1.0.
+ */
+package io.zeebe.transport.impl;
+
+import static io.zeebe.transport.impl.AtomixRequestSubscription.topicName;
+
+import io.zeebe.util.sched.future.CompletableActorFuture;
+import java.time.Duration;
+import java.util.function.Predicate;
+import java.util.function.Supplier;
+import org.agrona.DirectBuffer;
+
+final class RequestContext {
+
+  private final CompletableActorFuture<DirectBuffer> currentFuture;
+  private final Supplier<Integer> nodeIdSupplier;
+  private final int partitionId;
+  private final byte[] requestBytes;
+  private final long startTime;
+  private final Duration timeout;
+  private final Predicate<DirectBuffer> responseValidator;
+
+  RequestContext(
+      final CompletableActorFuture<DirectBuffer> currentFuture,
+      final Supplier<Integer> nodeIdSupplier,
+      final int partitionId,
+      final byte[] requestBytes,
+      final Predicate<DirectBuffer> responseValidator,
+      final Duration timeout) {
+    this.currentFuture = currentFuture;
+    this.nodeIdSupplier = nodeIdSupplier;
+    this.partitionId = partitionId;
+    this.requestBytes = requestBytes;
+    this.startTime = System.currentTimeMillis();
+    this.responseValidator = responseValidator;
+    this.timeout = timeout;
+  }
+
+  public boolean isDone() {
+    return currentFuture.isDone();
+  }
+
+  public CompletableActorFuture<DirectBuffer> getCurrentFuture() {
+    return currentFuture;
+  }
+
+  public Integer getNodeId() {
+    return nodeIdSupplier.get();
+  }
+
+  public String getTopicName() {
+    return topicName(partitionId);
+  }
+
+  public byte[] getRequestBytes() {
+    return requestBytes;
+  }
+
+  public Duration getTimeout() {
+    return timeout;
+  }
+
+  /**
+   * @return the time out, which is calculated via given timeout minus the already elapsed time.
+   *     this is necessary to respect the retries
+   */
+  public Duration calculateTimeout() {
+    final var currentTime = System.currentTimeMillis();
```

Comment:
I think we should use `ActorClock` instead.

Reply:
true you're right